snuba/snapshots/loaders/__init__.py

```diff
@@ -13,5 +14,5 @@ class BulkLoader(ABC):
     """
 
     @abstractmethod
-    def load(self, writer: BufferedWriterWrapper) -> None:
+    def load(self, writer: BufferedWriterWrapper, clickhouse_ro: ClickhousePool) -> None:
```

Comment:
I don't think we anticipated this as one of the impacted areas of this change, but this seems similar to the reader in that we'll want to find some sort of way to generalize this over time to break the dependency on the ClickHouse client directly.

Both the writer and the ClickHouse client here are derived the same writable storage when they are passed in from the command line tool. Should the writable storage be the thing that is passed here instead?

Reply:
Yeah, I think passing the writable storage here makes sense (at least for now).

Reply:
I did not yet find a good way to do this without introducing a circular dependency where the writer depends on the higher level storage.

Reply:
It's intentional in that I think many other changes need to happen here and I don't want to tackle it within the scope of this PR. 

- There are quite a number of places including this one where settings.CLICKHOUSE_HOST and friends are still being referenced, these all need to be removed of getting the information from the cluster. (optimize, cleanup, etc)
- I think we should probably just pass a storage into the bulk load command and remove the ability to pass a dataset and destination table. The dest_table comes from the storage.
- Also update confirm_load to do the same
- It depends on moving the writer (assuming we want to move it into storage set), which is not in scope of this PR

Reply:
I am ok holding off on this. Could you just mention with a TODO that this interface is going to change and in which direction ?